Index: SilverlightCompatability.cs
===================================================================
--- SilverlightCompatability.cs	(revision 2369)
+++ SilverlightCompatability.cs	(working copy)
@@ -211,6 +211,15 @@
 
             return results;
         }
+
+        public static int FindIndex<T>(this List<T> list, Predicate<T> match)
+        {
+            for (int i = 0; i < list.Count; i++)
+            {
+                if (match(list[i])) return i;
+            }
+            return -1;
+        }
     }
 }
 
