Index: SPARQLExpressionParser.cs
===================================================================
--- SPARQLExpressionParser.cs	(revision 2029)
+++ SPARQLExpressionParser.cs	(working copy)
@@ -1021,7 +1021,9 @@
                     }
 
                     //Return a Numeric Expression Term if it's an Integer/Decimal/Double
-                    if (XmlSpecsHelper.XmlSchemaDataTypeInteger.Equals(dtUri) && SparqlSpecsHelper.IsInteger(literal.Value))
+                    if ((XmlSpecsHelper.XmlSchemaDataTypeInteger.Equals(dtUri) ||
+                        XmlSpecsHelper.XmlSchemaDataTypeLong.Equals(dtUri))
+                        && SparqlSpecsHelper.IsInteger(literal.Value))
                     {
                         return new ConstantTerm(new LongNode(null, Int64.Parse(literal.Value)));
                     }
